pkgs/coverage/test/collect_coverage_config_test.dart

```diff
@@ -0,0 +1,213 @@
+    expect(testCoverage.functionCoverage, isTrue);
+    expect(testCoverage.branchCoverage, isFalse);
+    expect(testCoverage.scopeOutput, ['lib', 'src']);
+  });
+
+  group('partial yaml configuration', () {
+    test('override default values with partial yaml values 1', () async {
+      final configuredOptions = CoverageOptionsProvider(
+        filePath: 'test/test_coverage_options/partial_fields1.yaml',
+      ).coverageOptions;
+
+      // Parse arguments with empty command line args
+      final collectedCoverage = parseArgsCollectCoverage([], configuredOptions);
+      final testCoverage =
+          await parseArgsTestWithCoverage([], configuredOptions);
+      final formattedCoverage = parseArgsFormatCoverage([], configuredOptions);
+
+      expect(
+          collectedCoverage.outputDirectory,
+          getAbsolutePath(
+              'test/test_coverage_options/custom_coverage/coverage.json'));
+      expect(collectedCoverage.scopeOutput, ['lib', 'test']);
+      expect(collectedCoverage.functionCoverage, isFalse);
+      expect(
+          formattedCoverage.outputDirectory,
+          getAbsolutePath(
+              'test/test_coverage_options/custom_coverage/lcov.info'));
+      expect(testCoverage.packageName, 'Custom Dart Package');
+      expect(testCoverage.scopeOutput, ['lib', 'test']);
+    });
+
+    test('override default values with partial yaml values 2', () async {
+      final configuredOptions = CoverageOptionsProvider(
+        filePath: 'test/test_coverage_options/partial_fields2.yaml',
+      ).coverageOptions;
+
+      // Parse arguments with empty command line args
+      final collectedCoverage = parseArgsCollectCoverage([], configuredOptions);
+      final formattedCoverage = parseArgsFormatCoverage([], configuredOptions);
+      final testCoverage =
+          await parseArgsTestWithCoverage([], configuredOptions);
+
+      // Verify collect coverage yaml values
+      expect(collectedCoverage.scopeOutput, ['lib', 'tools']);
+      expect(collectedCoverage.branchCoverage, isFalse);
+      expect(collectedCoverage.functionCoverage, isTrue);
+      expect(
+          collectedCoverage.outputDirectory,
+          getAbsolutePath(
+              'test/test_coverage_options/custom_lcov/coverage.json'));
+
+      // Verify format coverage yaml values
+      expect(formattedCoverage.outputDirectory,
+          getAbsolutePath('test/test_coverage_options/custom_lcov/lcov.info'));
+      expect(formattedCoverage.packageDirectory,
+          getAbsolutePath('test/test_coverage_options'));
+
+      // Verify test with coverage yaml values
+      expect(testCoverage.packageName, 'coverage');
+      expect(testCoverage.outputDirectory,
+          getAbsolutePath('test/test_coverage_options/custom_lcov'));
+      expect(testCoverage.testScript, 'custom_test');
+      expect(testCoverage.functionCoverage, isTrue);
+    });
+  });
+
+  test('override yaml values with command line args', () async {
```

Comment:
Test coverage is generally pretty good in this file. The only other case I can see that needs to be tested is the way `collect_coverage` and `format_coverage` will output to stdout if you pass a literal `'stdout'` to the `--out` command line arg. Add a test for that case.